source/AI.cpp

```diff
@@ -158,10 +158,11 @@ namespace {
 	{
 		for(const Hardpoint &hardpoint : ship.Weapons())
 		{
-			const Weapon *weapon = hardpoint.GetOutfit();
-			if(weapon && !hardpoint.IsSpecial())
+			const Outfit *outfit = hardpoint.GetOutfit();
```

Comment:
Would it make sense to have a `GetWeapon()` and a `GetOutfit()` function on hardpoints? Because now we often seem to retrieve the outfit first, and then retrieve the weapon. Doing this directly within hardpoint would seem to make sense then.